src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -229,7 +231,7 @@ export class CodeActionModel extends Disposable {
 
 				const actions = createCancelablePromise(async token => {
 					if (this._settingEnabledNearbyQuickfixes() && trigger.trigger.type === CodeActionTriggerType.Invoke && (trigger.trigger.triggerAction === CodeActionTriggerSource.QuickFix || trigger.trigger.filter?.include?.contains(CodeActionKind.QuickFix))) {
-						const codeActionSet = await getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token);
+						const codeActionSet = this.disposables.add(await getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token));
```

Comment:
I think this introduces a leak as items are added to `disposables` but never removed

Instead try to make sure the old disposable values are released once a new set of values come in

Reply:
That sounds like something for a DisposableStore and/or a MutableDisposable